tools/tdx-measure/tdvf/tdvf.go

```diff
@@ -0,0 +1,138 @@
+	tdxMetadataOffsetGuid = []byte{0x35, 0x65, 0x7a, 0xe4, 0x4a, 0x98, 0x98, 0x47, 0x86, 0x5e, 0x46, 0x85, 0xa7, 0xbf, 0x8e, 0xc2}
+	metadataSignature     = "TDVF"
+)
+
+func findTdxMetadataOffset(firmare []byte) (uint32, error) {
+	footerGuid := firmare[len(firmare)-0x30:][:0x10]
+	if !bytes.Equal(footerGuid, tableFooterGuid) {
+		return 0, errors.New("can't find table footer GUID")
+	}
+
+	offset := len(firmare) - 0x32
+	tableLength := binary.LittleEndian.Uint16(firmare[offset:][:2])
+	endOffset := len(firmare) - 0x20 - int(tableLength)
+
+	for endOffset < offset {
+		entryBlockGuid := firmare[offset-0x10:][:0x10]
+		entryBlockLength := binary.LittleEndian.Uint16(firmare[offset-0x12:][:2])
+
+		if bytes.Equal(entryBlockGuid, tdxMetadataOffsetGuid) {
+			tdxOffset := binary.LittleEndian.Uint32(firmare[offset-0x16:][:4])
+			return tdxOffset, nil
+		}
+
+		offset -= int(entryBlockLength)
+	}
+
+	return 0, errors.New("can't find TDX metadata offset block entry")
+}
+
+type tdvfSection struct {
+	DataOffset     uint32
+	RawDataSize    uint32
+	MemoryAddress  uint64
+	MemoryDataSize uint64
+	Type           tdvfSectionType
+	Attributes     tdvfSectionAttributes
+}
+
+type tdvfSectionType uint32
+
+const (
+	Bbv tdvfSectionType = iota
+	Cfv
+	TdHob
+	TempMem
+	PermMem
+	Payload
+	PayloadParam
+	TdInfo
+)
+
+type tdvfSectionAttributes uint32
+
+const (
+	MrExtend tdvfSectionAttributes = 1 << iota
+	PageAug
+)
+
+func parseTdvfSections(firmare []byte) ([]tdvfSection, error) {
+	offset, err := findTdxMetadataOffset(firmare)
+	if err != nil {
+		return nil, fmt.Errorf("can't locate TDX firmware metadata offset: %w", err)
+	}
+
+	metadata := firmare[len(firmare)-int(offset):]
+	metadataHeader := metadata[:16]
+
+	if string(metadataHeader[:4]) != metadataSignature {
+		return nil, errors.New("unexpected signature")
+	}
+
+	length := binary.LittleEndian.Uint32(metadataHeader[4:][:4])
+	_ = length
+
+	version := binary.LittleEndian.Uint32(metadataHeader[8:][:4])
+	if version != 1 {
+		return nil, fmt.Errorf("expected version 1, got %v", version)
+	}
+
+	numberOfSectionEntries := binary.LittleEndian.Uint32(metadataHeader[12:][:4])
+	sections := make([]tdvfSection, numberOfSectionEntries)
+	sectionsData := metadata[16:]
+	for i := range numberOfSectionEntries {
+		sectionData := sectionsData[i*32:][:32]
+		section := &sections[i]
+		section.DataOffset = binary.LittleEndian.Uint32(sectionData[0:][:4])
+		section.RawDataSize = binary.LittleEndian.Uint32(sectionData[4:][:4])
+		section.MemoryAddress = binary.LittleEndian.Uint64(sectionData[8:][:8])
+		section.MemoryDataSize = binary.LittleEndian.Uint64(sectionData[16:][:8])
+		section.Type = tdvfSectionType(binary.LittleEndian.Uint32(sectionData[24:][:4]))
+		section.Attributes = tdvfSectionAttributes(binary.LittleEndian.Uint32(sectionData[28:][:4]))
+	}
+
+	return sections, nil
+}
+
+func CalculateMrTd(firmware []byte) ([48]byte, error) {
```

Comment:
nit: If would have expected func CalculateMrTd in mrtd.go xD

Reply:
This function is implemented as laid out in the TDVF spec (I'll add spec links). The specs containing MRTD specifics are guest agnostic and don't contain any suggestions on how to setup a guest. IMHO this function belongs here.